### PR TITLE
Added LocalDate and LocalDateTime TypeAdapters for Gson

### LINKED ISSUE #11: Issue serializing LocalDate and LocalDateTime
Trying to create an entity containing a LocalDate or LocalDateTime crashes with the exception:
```
com.google.gson.JsonIOException: Failed making field 'java.time.LocalDate#year' accessible; either increase its visibility or write a custom TypeAdapter for its declaring type.
```

I presume this is because Gson is used and Gson doesn't play well with java.time classes out of the box.

### PR DESCRIPTION
**Pull request by akaecliptic**
Added LocalDate and LocalDateTime TypeAdapters to resolve #11

### CODE REVIEW

**Review by phughk — COMMENTED** [review]
Thanks for submitting this change! Left some comments, but basically it would be cool if formats were configurable. Otherwise good and happy to proceed once that is resolved.

File: src/test/java/com/surrealdb/driver/SurrealDriverTest.java [hunk 1]
```diff
@@ -3,16 +3,16 @@
 import com.surrealdb.connection.SurrealConnection;
 import com.surrealdb.connection.SurrealWebSocketConnection;
 import com.surrealdb.connection.exception.SurrealRecordAlreadyExitsException;
-import com.surrealdb.driver.model.PartialPerson;
-import com.surrealdb.driver.model.Person;
-import com.surrealdb.driver.model.QueryResult;
+import com.surrealdb.driver.model.*;
```
[the remark this answers is not in the document]
Comment (akaecliptic) [reply src/test/java/com/surrealdb/driver/SurrealDriverTest.java:6]: Of course, that makes sense. Will fix this

File: src/main/java/com/surrealdb/connection/adapter/LocalDateTimeAdapter.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	public LocalDateTime read(JsonReader reader) throws IOException {
+		if (reader.peek() == JsonToken.NULL) {
+			reader.nextNull();
+			return null;
+		}
+
+		String date = reader.nextString();
+
+		try {
+			return (date.contains("Z") || date.contains("+")) ?
```
[the remark this answers is not in the document]
Comment (akaecliptic) [reply src/main/java/com/surrealdb/connection/adapter/LocalDateTimeAdapter.java:37]: Hmm. I was referencing [this section on datetime](https://surrealdb.com/docs/surrealql/datamodel/datetimes) from the SurrealDB documentation, it looks like conversion of short hand yyyy-MM-DD to the full ISO-8601 is done by the database (please, do correct me if I'm wrong). 

So, I went with this "simple" approach. I agree about using a formatter, and I'll try make that change. Not sure the best approach about custom formatter though. Thanks

**Comment by akaecliptic** [comment]
Still thinking over the cleanest way to add custom adapters and date formatters. Will come back to that later.

**Pull request merged**